extensions/azdata/src/test/azdata.test.ts

```diff
@@ -52,67 +49,20 @@ describe('azdata', function () {
 		it('successful install', async function (): Promise<void> {
 			switch (process.platform) {
 				case 'win32':
-					{
-						const executeCommandStub = sinon.stub(childProcess, 'executeCommand').callsFake(async (command: string, args: string[]) => {
-							should(command).be.equal('msiexec');
-							should(args[0]).be.equal('/qn');
-							should(args[1]).be.equal('/i');
-							should(path.basename(args[2])).be.equal(azdata.azdataUri);
-							return { stdout: 'success', stderr: '' };
-						});
-						nock(azdata.azdataHostname)
-							.get(`/${azdata.azdataUri}`)
-							.replyWithFile(200, __filename);
-						await azdata.installAzdata(outputChannelMock.object);
-
-						should(executeCommandStub.calledOnce).be.true();
-					}
+					await testWin32SuccessfulInstall();
 					break;
 				case 'darwin':
-					{
-						const executeCommandStub = sinon.stub(childProcess, 'executeCommand').callsFake(async (command: string, _args: string[]) => {
-							should(command).be.equal('brew');
-							return { stdout: 'success', stderr: '' };
-						});
-						await azdata.installAzdata(outputChannelMock.object);
-						should(executeCommandStub.calledThrice).be.true();
-					}
+					await testDarwinSuccessfulInstall();
 					break;
 				case 'linux':
-					{
-						const executeCommandStub = sinon.stub(childProcess, 'executeCommand').callsFake(async (command: string, _args: string[]) => {
-							should(command).be.equal('lsb_release');
-							return { stdout: 'success', stderr: '' };
-						});
-						let callNumber = 0;
-						const executeSudoCommandStub = sinon.stub(childProcess, 'executeSudoCommand').callsFake(async (command: string) => {
-							++callNumber;
-							switch (callNumber) {
-								case 3:
-									should(command).match(/^curl/);
-									break;
-								case 4:
-									should(command).match(/^add-apt-repository/);
-									break;
-								default:
-									should(command).match(/^apt-get/);
-									break;
-							}
-							return { stdout: 'success', stderr: '' };
-						});
-						await azdata.installAzdata(outputChannelMock.object);
-						should(executeSudoCommandStub.callCount).be.equal(6);
-						should(executeCommandStub.calledOnce).be.true();
-					}
+					await testLinuxSuccessfulInstall();
 					break;
 			}
 		});
 
 		if (process.platform === 'win32') {
 			it('unsuccessful download - win32', async function (): Promise<void> {
-				nock(azdata.azdataHostname)
-					.get(`/${azdata.azdataUri}`)
-					.reply(404);
+				sinon.stub(HttpClient, 'downloadFile').rejects();
```

Comment:
Why did you change this? I would prefer we keep the test handling the actual HTTP errors.

Reply:
for isolation, HttpClient.test.ts already tests HttpClient APIs work correctly so just mocking at HttpClient level gives azdata tests isolation so that if HttpClient's implementation changes without corresponding api change then azdata tests are not affected.

Reply:
I kind of would have rather kept the logic and then just had a helper function that set up the mocks that both called. We don't need to completely isolate everything from everything else - especially when the code is part of the same module.

Your reasoning here isn't bad - I just think this change was unnecessary and isn't really helping anything except make more assumptions about how downloadFile behaves.